examples/spark/spark-gluster/README.md

```diff
@@ -0,0 +1,163 @@
+
+```shell
+$ kubectl create -f examples/spark/spark-gluster/spark-master-service.yaml
+```
+
+Verify that the Spark Master Service deployed successfully.
+
+```shell
+$ kubectl get services
+```
+
+## Step Three: Start your Spark workers
+
+Modify the `examples/spark/spark-gluster/spark-worker-rc.yaml` file to reflect the GlusterFS Volume that you wish to use in the PATH parameter of the Volumes subsection.
+
+Make sure that the replication factor for the pods is not greater than the amount of Kubernetes nodes available in your Kubernetes cluster.
+
+Submit your Spark Worker ReplicationController by running the following command:
+
+```shell
+$ kubectl create -f examples/spark/spark-gluster/spark-worker-rc.yaml
+```
+
+Verify that the Spark Worker ReplicationController deployed its pods successfully.
+
+```shell
+$ kubectl get pods
+```
+
+Follow the steps from the standard example to verify the Spark Worker pods have registered successfully with the Spark Master.
+
+## Step Four: Submit a Spark Job
+
+All the Spark Workers and the Spark Master in your cluster have a mount to GlusterFS. This means that any of them can be used as the Spark Client to submit a job. For simplicity, lets use the Spark Master as an example.
+
+Obtain the Host (Kubernetes Node) that the Spark Master container is running on
+
+```shell
+$ kubectl describe pod spark-master
+```
+
+Shell into the Kubernetes Node running the Spark Worker:
+
+```shell
+$ ssh root@worker-1
+```
+
+Identify the Container ID for the Spark Master:
+
+```
+$ docker ps
+CONTAINER ID        IMAGE                                  COMMAND             CREATED             STATUS              PORTS                    NAMES
+88a8531f9329        gcr.io/google_containers/spark-master:latest              "/start.sh"         4 minutes ago       Up 4 minutes                                 k8s_spark-master.af6b2d08_sgc-sfgj0_default_446a9f27-e8a2-11e4-ad4a-000c29151bdb_b2710ef5   
+4a58e1f3489b        gcr.io/google_containers/pause:0.8.0   "/pause"            4 minutes ago       Up 4 minutes        0.0.0.0:8888->8088/tcp   k8s_POD.908f04ee_sgc-sfgj0_default_446a9f27-e8a2-11e4-ad4a-000c29151bdb_895cdc58 
+```
+
+Now we will shell into the Spark Master Container and run a Spark Job. In the example below, we are running the Spark Wordcount example and specifying the input and output directory at the location where GlusterFS is mounted in the Spark Master Container. This will submit the job to the Spark Master who will distribute the work to all the Spark Worker Containers. All the Spark Worker containers  will be able to access the data as they all have the same GlusterFS volume mounted at /mnt/glusterfs. The reason we are submitting the job from a Spark Worker and not an additional Spark Base container (as in the standard Spark Example) is due to the fact that the Spark instance submitting the job must be able to access the data. Only the Spark Master and Spark Worker containers have GlusterFS mounted.
```

Comment:
+1 for `kubectl exec` (https://github.com/kubernetes/kubernetes/tree/master/examples/spark#step-four-do-something-with-the-cluster)

> The reason we are submitting the job from a Spark Worker and not an additional Spark Base container

Aren't you submitting the job to the master? What is the "Spark Base container" (is that the same as the Driver)?

_Aside: @bgrant0607 do we have any guidelines about line length? It would be easier to comment on specific parts of this paragraph if it was multiple lines._